### PR TITLE
Add headers for CopyFromUri & StartCopyFromUri for BlobClient

### PR DESCRIPTION
**Pull request by yah01**
For CopyFromUri & StartCopyFromUri method of BlockBlob:

Add header [x-ms-blob-type: BlockBlob"]
Add header [x-ms-copy-source-blob-properties], default by true

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp [hunk 1]
```diff
@@ -495,6 +495,11 @@ namespace Azure { namespace Storage { namespace Blobs {
      * @brief If the destination blob should be sealed. Only applicable for Append Blobs.
      */
     Azure::Nullable<bool> ShouldSealDestination;
+
+    /**
+     * @brief If the destination blob should copy the properties of the source
+     */
+    Azure::Nullable<bool> ShouldCopySourceBlobProperties;
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:502]: It looks to me this value should be either true or false. What does null mean here? What will happen the HTTP header is not included in the request?

File: sdk/storage/azure-storage-blobs/src/blob_client.cpp [hunk 2]
```diff
@@ -560,6 +562,11 @@ namespace Azure { namespace Storage { namespace Blobs {
       protocolLayerOptions.TransactionalContentHash = options.TransactionalContentHash;
     }
 
+    if (!options.ShouldCopySourceBlobProperties.HasValue())
+    {
+      protocolLayerOptions.ShouldCopySourceBlobProperties = Azure::Nullable<bool>(true);
+    }
+
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/src/blob_client.cpp:569]:
```suggestion
    protocolLayerOptions.ShouldCopySourceBlobProperties = options.ShouldCopySourceBlobProperties;
```

File: sdk/storage/azure-storage-blobs/src/blob_rest_client.cpp [hunk 3]
```diff
@@ -1,100 +1,100 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-blobs/src/blob_rest_client.cpp:1]: The file content is not changed at all.

**Review by Jinming-Hu — CHANGES_REQUESTED** [review]
Add unit test

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp (hunk 1 above)
Comment (yah01) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:502]: It's an optional header, it's default by true if not included in HTTP header

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:502]: Then I guess this should not be nullable.

```suggestion
    bool ShouldCopySourceBlobProperties = true;
```

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp [hunk 4]
```diff
@@ -495,6 +495,11 @@ namespace Azure { namespace Storage { namespace Blobs {
      * @brief If the destination blob should be sealed. Only applicable for Append Blobs.
      */
     Azure::Nullable<bool> ShouldSealDestination;
+
+    /**
+     * @brief Whether copy the properties of source to the destition, it's default by true
```
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:500]: typo for : destition

```suggestion
     * @brief Whether copy the properties of source to the destination, it's true by default.
```

**Comment by Jinming-Hu** [comment]
New headers doesn't work, close it for now

**Pull request closed without merging**